src/wp-includes/media.php

```diff
@@ -978,23 +979,23 @@ function wp_get_attachment_image_srcset( $attachment_id, $size = 'medium' ) {
  *
  * @since 4.4.0
  *
- * @param int    $attachment_id Image attachment ID.
- * @param string $size          Optional. Name of image size. Default value: 'medium'.
- * @param array  $args {
- *     Optional. Arguments to retrieve attachments.
- *
- *     @type array|string $sizes An array or string containing of size information.
- *     @type int          $width A single width value used in the default `sizes` string.
- * }
+ * @param int          $attachment_id Image attachment ID.
+ * @param string|array $size          Optional. Image size name or a flat array
+ *                                    of width and height values. Default 'medium'.
+ * @param int          $width         Optional. A single width value used to
+ *                                    create the default value of the 'sizes' attribute.
  * @return string|bool A valid source size value for use in a 'sizes' attribute or false.
  */
-function wp_get_attachment_image_sizes( $attachment_id, $size = 'medium', $args = null ) {
+function wp_get_attachment_image_sizes( $attachment_id, $size = 'medium', $width = null ) {
 
-	// Try to get the image width from $args first.
-	if ( is_array( $args ) && ! empty( $args['width'] ) ) {
-		$img_width = (int) $args['width'];
-	} elseif ( $img = image_get_intermediate_size( $attachment_id, $size ) ) {
-		list( $img_width, $img_height ) = image_constrain_size_for_editor( $img['width'], $img['height'], $size );
+	// Try to get the image width from $width or $size first.
+	if ( $width ) {
+		$img_width = intval( $width );
+	} elseif ( is_array( $size ) ) {
```

Comment:
@joemcgill 

I understand that if ```$size``` is an array of width and height values instead of an image name there is probably not a crop with those dimensions so a larger image will be used. But can't we assume that the width in the array is the width at which the image will be displayed? That's why I thought it would be better to use the width from the array instead of the one from ```image_get_intermediate_size``` (besides saving the call to that function).